cognite/neat/rules/models/_rules/domain_rules.py

```diff
@@ -32,93 +18,17 @@ class DomainMetadata(BaseMetadata):
-        else:
-            return value
-
-    @field_validator("property_", mode="after")
-    def is_property_id_compliant(cls, value):
-        if re.search(more_than_one_none_alphanumerics_regex, value):
-            raise exceptions.MoreThanOneNonAlphanumericCharacter("property_id", value).to_pydantic_custom_error()
-        if not re.match(property_id_compliance_regex, value):
-            raise exceptions.PropertyIDRegexViolation(value, property_id_compliance_regex).to_pydantic_custom_error()
-        else:
-            return value
-
 
 class DomainClass(SheetEntity):
-    class_: str = Field(alias="Class")
+    class_: Class_ = Field(alias="Class")
     description: str | None = Field(None, alias="Description")
-    parent: list[ParentClass] | None = Field(alias="Parent Class", default=None)
-
-    @field_validator("parent", mode="before")
-    def parent_class_to_list_of_entities(cls, value, values):
-        if isinstance(value, str) and value:
-            parents = []
-            for v in value.replace(", ", ",").split(","):
-                if ENTITY_ID_REGEX_COMPILED.match(v) or VERSIONED_ENTITY_REGEX_COMPILED.match(v):
-                    parents.append(ParentClass.from_string(entity_string=v))
-                else:
-                    # if all fails defaults "neat" object which ends up being updated to proper
-                    # prefix and version upon completion of Rules validation
-                    parents.append(ParentClass(prefix="undefined", suffix=v, name=v))
-
-            return parents
-        else:
-            return None
-
-    @field_validator("class_", mode="after")
-    def is_class_id_compliant(cls, value):
-        if re.search(more_than_one_none_alphanumerics_regex, value):
-            raise exceptions.MoreThanOneNonAlphanumericCharacter("class_id", value).to_pydantic_custom_error()
-        if not re.match(class_id_compliance_regex, value):
-            raise exceptions.PropertiesSheetClassIDRegexViolation(
-                value, class_id_compliance_regex
-            ).to_pydantic_custom_error()
-        else:
-            return value
-
-    @field_validator("parent", mode="after")
-    def is_parent_class_id_compliant(cls, value):
-        if isinstance(value, list):
-            if illegal_ids := [v for v in value if re.search(more_than_one_none_alphanumerics_regex, v.suffix)]:
-                raise exceptions.MoreThanOneNonAlphanumericCharacter(
-                    "parent", ", ".join(illegal_ids)
-                ).to_pydantic_custom_error()
-            if illegal_ids := [v for v in value if not re.match(class_id_compliance_regex, v.suffix)]:
-                for v in illegal_ids:
-                    print(v.id)
-                raise exceptions.ClassSheetParentClassIDRegexViolation(
-                    illegal_ids, class_id_compliance_regex
-                ).to_pydantic_custom_error()
-        return value
+    parent: ParentClass_ = Field(alias="Parent Class")
```

Comment:
This does not collide with Python keyword, so no need to suffix and underscore. 
```suggestion
    parent: ParentClass = Field(alias="Parent Class")
```

Reply:
It collides with `ParentClass` , which is a class, while `ParentClass_` is a type we define.